<a href="https://colab.research.google.com/github/Chandana-madarapu/Data_science/blob/main/Statistics2_Confidence_Interval.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
import scipy.stats as st

In [ ]:
# Sample data
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25,
                 1.36, 1.32, 0.85, 1.07, 1.48,
                 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)                 # sample size
mean = np.mean(data)          # sample mean
s = np.std(data, ddof=1)      # sample standard deviation


In [ ]:
alpha = 0.01
df = n - 1
SE = s / np.sqrt(n)

t_crit = st.t.ppf(1 - alpha/2, df)

ci_t = (mean - t_crit*SE, mean + t_crit*SE)

print(ci_t)


(np.float64(1.0901973384384906), np.float64(1.3871359948948425))


In [ ]:
sigma = 0.2   # given population std
SE_known = sigma / np.sqrt(n)

z_crit = st.norm.ppf(1 - alpha/2)

ci_z = (mean - z_crit*SE_known, mean + z_crit*SE_known)

print(ci_z)


(np.float64(1.1056514133957607), np.float64(1.3716819199375725))


Both confidence intervals are almost identical, which indicates that the estimated mean durability of the print-heads is consistent and reliable. The interval calculated using the t-distribution is slightly wider than the one obtained from the z-distribution because the population standard deviation is unknown, and using the sample standard deviation adds a small amount of uncertainty. The t-distribution adjusts for this by widening the interval slightly to account for that uncertainty. Overall, it can be concluded with 99% confidence that the true average durability of the print-heads lies between approximately 1.09 and 1.39 million characters before failure